device.go

```diff
@@ -45,6 +45,10 @@ type DeviceDesc struct {
 
 	// Configuration information
 	Configs map[int]ConfigDesc
+
+	IdxSerial       int   // The index into the getStringDescriptor() to get the Serial Number
```

Comment:
I don't know whether it is possible to extract the data during enumeration. It might be necessary to have the device open to access the data (not sure, though, I didn't try).

The 2nd approach will work for sure. The wrapper methods could be in place in any way, even if the 1st approach turns out to be working. On the API there is a function to be called that either accesses the already-stored data (1st approach) or gathers the information when being called (2nd approach).

Sorry for the delayed answer.